packages/FormElement/src/FormElement.js

```diff
@@ -71,85 +72,73 @@ function FormElement(props) {
+  const hasError = !!extractedChildren["FormElement.Error"] && !!extractedChildren["FormElement.Error"].props.children;
   const uniqueInputId = React.useRef(uuidv4()).current;
   const inputId = isNil(id) || id === "" ? uniqueInputId : id;
-  let isFooterInserted = false;
 
   function renderFooter() {
-    if (isFooterInserted) return;
-
-    isFooterInserted = true;
-
     if (hasError) {
-      return extratedChildren["FormElement.Error"];
+      return extractedChildren["FormElement.Error"];
     }
 
-    if (extratedChildren["FormElement.Description"]) {
-      return React.cloneElement(extratedChildren["FormElement.Description"], {
+    if (extractedChildren["FormElement.Description"]) {
+      return React.cloneElement(extractedChildren["FormElement.Description"], {
         ariaDescriptionId,
       });
     }
 
     return null;
   }
 
+  const childAttributes = {
+    className: "form-element--child",
+    "aria-describedby": ariaDescriptionId,
+    id: inputId,
+  };
+
   return (
     <div css={formElementStyles} isInline={isInline} size={size} isDisabled={isDisabled} {...moreProps}>
       <Label
         hasOptionalLabel={hasOptionalLabel}
         hasRequiredLabel={hasRequiredLabel}
-        help={extratedChildren["FormElement.Help"]}
+        help={extractedChildren["FormElement.Help"]}
         id={inputId}
         isInline={isInline}
         isVisuallyHidden={isLabelVisuallyHidden}
         label={label}
       />
-
-      {extratedChildren.children.map(child => {
-        if (React.isValidElement(child)) {
-          const extendedProps = isString(child.type)
-            ? {
-                "aria-describedby": ariaDescriptionId,
-                id: inputId,
-                disabled: isDisabled,
-                readOnly: isReadOnly,
-              }
-            : {
-                "aria-describedby": ariaDescriptionId,
-                hasError,
-                id: inputId,
-                isDisabled,
-                isReadOnly,
-                size,
-              };
-          const clonedChild = React.cloneElement(child, extendedProps);
-
-          return (
-            <React.Fragment key={child.key}>
-              {isInline ? (
-                <div css={inlineContainerStyles}>
-                  {clonedChild}
-                  {renderFooter()}
-                </div>
-              ) : (
-                <React.Fragment>
-                  {clonedChild}
-                  {renderFooter()}
-                </React.Fragment>
-              )}
-            </React.Fragment>
-          );
-        }
-
-        return child;
-      })}
+      <div css={isInline ? inlineContainerStyles : null}>
+        {extractedChildren["FormElement.ExtraPanel"] ? extractedChildren["FormElement.ExtraPanel"] : null}
+
+        {extractedChildren.children.map(child => {
+          if (React.isValidElement(child)) {
+            const extendedProps = isString(child.type)
```

Comment:
It's not obvious what condition you really care about with `isString(child.type)`.  Can we do something like:
`const isNativeFormControl = isString(child.type);` to improve readability?
cc @allison-c